docs/standard/commandline/model-binding.md

```diff
@@ -149,13 +149,17 @@ Because `AllowMultipleArgumentsPerToken` is set to `true`, the following input r
 
 ### File system types
 
-Since command line applications often have to work with the file system, `FileInfo` and `DirectoryInfo` are supported. The following example shows the use of `FileInfo`, but it would work also with `DirectoryInfo`.
+Command line applications that work with the file system can use the <xref:System.IO.FileSystemInfo>, <xref:System.IO.FileInfo>, and <xref:System.IO.DirectoryInfo> types. The following example shows the use of `FileSystemInfo`:
 
-:::code language="csharp" source="snippets/model-binding/csharp/Program.cs" id="fileinfo" :::
+:::code language="csharp" source="snippets/model-binding/csharp/Program.cs" id="filesysteminfo" :::
+
+With `FileInfo` and `DirectoryInfo` the pattern matching code is not required:
+
+:::code language="csharp" source="snippets/model-binding/csharp/Program.cs" id="filesysteminfo" :::
 
 ### Anything with a string constructor
 
-`FileInfo` and `DirectoryInfo` aren't special cases. Any type having a constructor that takes a single string parameter can be bound in this way. For example, the same code works with a `Uri` instead.
+The file system types aren't special cases. Any type having a constructor that takes a single string parameter can be bound in this way. For example, code that would work with `FileInfo` works with a <xref:System.Uri> instead.
```

Comment:
This is no longer true after the work to support NativeAOT compilation, unfortunately. These are special cases. (`System.Uri` is another, and I expect the list might grow over time).

Reply:
Are these now the only special cases or are there others that we should mention?

Reply:
Technically, pretty much all types are now special cases. The two main conventions we used before doing the trimming work were the single-string constructor convention and the presence of a `TypeConverter`. Removing the latter also meant types like `DateTime` and `int` needed to be accounted for with special cases. You can see the implementations here: https://github.com/dotnet/command-line-api/blob/021ec68a4cb510c2cc125c6ebb78b9cfd4e3847a/src/System.CommandLine/Binding/ArgumentConverter.StringConverters.cs#L11